packages/signers/src/offchain-message-signer.ts

```diff
@@ -0,0 +1,111 @@
+import { OffchainMessage, OffchainMessageSignatory } from '@solana/offchain-messages';
+
+import { deduplicateSigners } from './deduplicate-signers';
+import { MessageSigner } from './message-signer';
+
+/**
+ * An extension of the {@link OffchainMessageSignatory} type that allows us to store
+ * {@link MessageSigner | MessageSigners} inside it.
+ *
+ * @typeParam TAddress - Supply a string literal to define an account having a particular address.
+ * @typeParam TSigner - Optionally provide a narrower type for the {@link MessageSigner} to use.
+ *
+ * @interface
+ *
+ * @example
+ * ```ts
+ * import { generateKeyPairSigner, OffchainMessageSignatoryWithSigner } from '@solana/signers';
+ *
+ * const signer = await generateKeyPairSigner();
+ * const messageSignatory: OffchainMessageSignatoryWithSigner = {
+ *     address: signer.address,
+ *     signer,
+ * };
+ * ```
+ */
+export interface OffchainMessageSignatoryWithSigner<
+    TAddress extends string = string,
+    TSigner extends MessageSigner<TAddress> = MessageSigner<TAddress>,
+> extends OffchainMessageSignatory<TAddress> {
+    readonly signer: TSigner;
+}
```

Comment:
Sorry if this is still WIP and going to change but shouldn't this be something like:

```ts
type OffchainMessageSignatoryWithSigner<TAddress> = OffchainMessageSignatory<TAddress> & MessageSigner<TAddress>;
```

So that the signatory _is_ a `Signer` instead of containing one.